nixos/lib/make-disk-image.nix

```diff
@@ -405,7 +477,17 @@ let format' = format; in let
       ''}
 
       ${lib.optionalString installBootLoader ''
-        # Set up core system link, GRUB, etc.
+        # In this throwaway resource, we only have /dev/vda, but the actual VM may refer to another disk for bootloader, e.g. /dev/vdb
+        # Use this option to create a symlink from vda to any arbitrary device you want.
+        ${optionalString (config.boot.loader.grub.device != "/dev/vda") ''
+            ln -s /dev/vda ${config.boot.loader.grub.device}
+        ''}
+        # TODO: needed?
+        # For GRUB 0.97 compatibility
+        mkdir -p /mnt/boot/grub
+        echo '(hd0) /dev/vda' > /mnt/boot/grub/device.map
```

Comment:
Not sure what the plans are to yet out grub without 2.

Reply:
I think we are safe to remove it, but wanted to get some opinions on this. No one complained for some time now, I guess this can be removed and we can revert if it breaks boot for someone.